### PR TITLE
Remove the ability to configure the user config path

### PR DESCRIPTION
**Pull request by Veykril**
Being able to do this makes little sense as this is effectively a cyclic dependency (and we do not want to fixpoint this really).

### CODE REVIEW

File: crates/rust-analyzer/src/config.rs [hunk 1]
```diff
@@ -794,8 +786,27 @@ impl std::ops::Deref for Config {
...
+    /// | Linux   | `$XDG_CONFIG_HOME` or `$HOME`/.config | /home/alice/.config                      |
+    /// | macOS   | `$HOME`/Library/Application Support   | /Users/Alice/Library/Application Support |
+    /// | Windows | `{FOLDERID_RoamingAppData}`           | C:\Users\Alice\AppData\Roaming           |
+    pub fn user_config_path() -> &'static AbsPath {
+        static USER_CONFIG_PATH: LazyLock<AbsPathBuf> = LazyLock::new(|| {
+            let user_config_path = if let Some(path) = env::var_os("__TEST_RA_USER_CONFIG_DIR") {
+                std::path::PathBuf::from(path)
+            } else {
+                dirs::config_dir()
+                    .expect("A config dir is expected to existed on all platforms ra supports.")
```
Comment (lnicola) [inline crates/rust-analyzer/src/config.rs:803]: This feels a bit too heavy-handed. I don't think we should crash `~/.config` doesn't exist, it doesn't get created until someone actually saves a file there.

Comment (lnicola) [reply crates/rust-analyzer/src/config.rs:803]: Ah, never mind, it doesn't check for existence.

Comment (Veykril) [reply crates/rust-analyzer/src/config.rs:803]: The panic message is misleading 😅

Comment (lnicola) [reply crates/rust-analyzer/src/config.rs:803]: I'm still not convinced that not having the directory should be a fatal error. r-a can still work fine enough without one. The Rust toolchain might not, but that's not our problem.

Comment (lnicola) [reply crates/rust-analyzer/src/config.rs:803]: There's no config dir on WASM, for example.

**Review by alibektas — APPROVED** [review]

**Pull request merged**